src/lib/DefaultRuleManager.ts

```diff
@@ -200,7 +200,16 @@ export class DefaultRuleManager implements RuleManager {
 		await Lifecycle.getInstance().emitTelemetry(runTelemetryObject);
 	}
 
-	protected async resolveEngineFilters(filters: RuleFilter[], engineOptions: Map<string,string> = new Map()): Promise<RuleEngine[]> {
+	/**
+	 * Returns a list of engines that match the provided filter criteria.
+	 * Additionally, if a {@link RunOptions} object is provided, then only engines
+	 * eligible to run under those options will be returned.
+	 * @param filters
+	 * @param engineOptions
+	 * @param [runOptions]
+	 * @protected
+	 */
+	protected async resolveEngineFilters(filters: RuleFilter[], engineOptions: Map<string,string> = new Map(), runOptions?: RunOptions): Promise<RuleEngine[]> {
```

Comment:
`runOptions` is an optional parameter to allow the `list` flow to still call this method.